xformers/components/attention/csrc/cuda/mem_eff_attention/attention_backward_generic.cu

```diff
@@ -100,11 +100,25 @@ mem_efficient_attention_backward_cutlass(
   // keys with no query associated, so they are not
   // initialized
   bool grad_kv_needs_init = causal && N > M;
-  at::Tensor grad_q = at::empty_like(query);
-  at::Tensor grad_k =
-      grad_kv_needs_init ? at::zeros_like(key) : at::empty_like(key);
-  at::Tensor grad_v =
-      grad_kv_needs_init ? at::zeros_like(value) : at::empty_like(value);
+  at::Tensor grad_q, grad_k, grad_v;
+  if (!grad_kv_needs_init && query.size(1) == key.size(1) &&
```

Comment:
nit: looks like the `!grad_kv_needs_init` is redundant as `N > M` is always false if `query.size(1) == key.size(1)` ?